django/db/models/base.py

```diff
@@ -1780,13 +1780,9 @@ def _check_constraints(cls):
 def method_set_order(self, ordered_obj, id_list, using=None):
     if using is None:
         using = DEFAULT_DB_ALIAS
-    order_wrt = ordered_obj._meta.order_with_respect_to
-    filter_args = order_wrt.get_forward_related_filter(self)
```

Comment:
I was surprised no tests fail when ignoring `filter_args`. Not sure if we should keep filtering the queryset by it or simply ignore it at this point.

Reply:
It looks like this has been around forever 9c52d56f6f8a9cdafb231adf9f4110473099c9b5

Reply:
Should we remove the same from `method_get_order()`? :thinking:

Reply:
@felixxm I think it's required there because of we can't filter by ids.

Reply:
:man_facepalming: Ah, ok. Thanks :+1:

Reply:
I checked and this logic (in other form) is there since 2005. I don't think it's required.

Reply:
From looking at the documentation it looks like the filter is still required to prevent `question.set_answer_order()` calls with `Answer.id`s that are not associated with `question` from being altered as well. I'll add a regression test for this case in a different commit to make sure we don't regress here on in the future.

Reply:
I added the test as Simon suggested.